internal/verifier/verifier.go

```diff
@@ -190,18 +190,31 @@ type ProvenanceIRVerifier struct {
 // Verify verifies an instance of ProvenanceIRVerifier by comparing its Got and Want fields.
 // All empty fields are ignored. If a field in Got contains more than one value, we return an error.
 func (verifier *ProvenanceIRVerifier) Verify() (VerificationResult, error) {
-	result := NewVerificationResult()
-	if len(verifier.Got.BinarySHA256Digests) != 1 {
-		return result, fmt.Errorf("provenance must have exactly one binary SHA256 digest value, got (%v)", verifier.Got.BinarySHA256Digests)
+	combinedResult := NewVerificationResult()
+
+	// Verify BinarySHA256 Digest.
+	if verifier.Want.BinarySHA256Digests != nil {
+		if len(verifier.Got.BinarySHA256Digests) != 1 {
+			return combinedResult, fmt.Errorf("provenance must have exactly one binary SHA256 digest value, got (%v)", verifier.Got.BinarySHA256Digests)
+		}
+		nextResult, err := verifier.Got.verifyBinarySHA256Digest(verifier.Want)
+		if err != nil {
+			return combinedResult, fmt.Errorf("provenance must have exactly one binary SHA256 digest value, got (%v)", verifier.Got.BinarySHA256Digests)
+		}
+		combinedResult.Combine(nextResult)
 	}
 
-	nextresult, err := verifier.Got.verifyBinarySHA256Digest(verifier.Want)
-	if err != nil {
-		return result, fmt.Errorf("provenance must have exactly one binary SHA256 digest value, got (%v)", verifier.Got.BinarySHA256Digests)
+	// Verify HasBuildCmd.
+	if verifier.Want.BuildCmds != nil {
```

Comment:
When we load a toml file of reference values, which does not have a build command, is `BuildCmds` set to `nil` or empty array?